ioctl/newcmd/hdwallet/hdwalletderive.go

```diff
@@ -8,46 +8,94 @@ package hdwallet
 
+	"github.com/iotexproject/iotex-core/ioctl"
 	"github.com/iotexproject/iotex-core/ioctl/config"
-	"github.com/iotexproject/iotex-core/ioctl/output"
 	"github.com/iotexproject/iotex-core/ioctl/util"
 	"github.com/iotexproject/iotex-core/pkg/util/fileutil"
 )
 
+// Multi-language support
+var (
+	_hdwalletDeriveCmdUses = map[config.Language]string{
+		config.English: "derive id1/id2/id3",
+		config.Chinese: "derive id1/id2/id3",
+	}
+	_hdwalletDeriveCmdShorts = map[config.Language]string{
+		config.English: "derive key from HDWallet",
+		config.Chinese: "查询HDWallet钱包的派生key地址",
+	}
+)
+
+// NewHdwalletDeriveCmd represents the hdwallet derive command
+func NewHdwalletDeriveCmd(client ioctl.Client) *cobra.Command {
+	use, _ := client.SelectTranslation(_hdwalletDeriveCmdUses)
+	short, _ := client.SelectTranslation(_hdwalletDeriveCmdShorts)
+
+	cmd := &cobra.Command{
+		Use:   use,
+		Short: short,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			signer := "hdw::" + args[0]
+			account, change, index, err := util.ParseHdwPath(signer)
+			if err != nil {
+				return errors.New("invalid hdwallet key format")
+			}
+
+			cmd.Println("Enter password\n")
+			password, err := client.ReadSecret()
+			if err != nil {
+				return errors.New("failed to get password")
+			}
+
+			addr, _, err := DeriveKey(account, change, index, password)
+			if err != nil {
+				return err
+			}
+			cmd.Println(fmt.Sprintf("address: %s\n", addr))
+			return nil
+		},
+	}
+	return cmd
+}
+
 // DeriveKey derives the key according to path
 func DeriveKey(account, change, index uint32, password string) (string, crypto.PrivateKey, error) {
 	// derive key as "m/44'/304'/account'/change/index"
 	hdWalletConfigFile := config.ReadConfig.Wallet + "/hdwallet"
 	if !fileutil.FileExists(hdWalletConfigFile) {
-		return "", nil, output.NewError(output.InputError, "Run 'ioctl hdwallet create' to create your HDWallet first.", nil)
+		return "", nil, errors.New("run 'ioctl hdwallet create' to create your HDWallet first")
 	}
 
 	enctxt, err := os.ReadFile(hdWalletConfigFile)
 	if err != nil {
-		return "", nil, output.NewError(output.InputError, "failed to read config", err)
+		return "", nil, errors.New("failed to read config")
 	}
 
 	enckey := util.HashSHA256([]byte(password))
 	dectxt, err := util.Decrypt(enctxt, enckey)
 	if err != nil {
-		return "", nil, output.NewError(output.InputError, "failed to decrypt", err)
+		return "", nil, errors.New("failed to decrypt")
 	}
 
 	dectxtLen := len(dectxt)
 	if dectxtLen <= 32 {
-		return "", nil, output.NewError(output.ValidationError, "incorrect data", nil)
+		return "", nil, errors.New("incorrect data")
 	}
 
 	mnemonic, hash := dectxt[:dectxtLen-32], dectxt[dectxtLen-32:]
 	if !bytes.Equal(hash, util.HashSHA256(mnemonic)) {
-		return "", nil, output.NewError(output.ValidationError, "password error", nil)
+		return "", nil, errors.New("password error")
 	}
 
```

Comment:
split to two function, and move line75~99 into client interface.  consider [#3423 ](https://github.com/iotexproject/iotex-core/pull/3423) together